gammapy/scripts/analysis.py

```diff
@@ -51,6 +60,124 @@ def settings(self):
+        self.config.validate()
+
+        datastore_path = make_path(self.settings["observations"]["datastore"])
+        if datastore_path.is_file():
+            self.datastore = DataStore().from_file(datastore_path)
+        elif datastore_path.is_dir():
+            self.datastore = DataStore().from_dir(datastore_path)
+        else:
+            log.error("Datastore {} not found.".format(datastore_path))
+            return False
+
+        ids = set()
+        selection = dict()
+        for criteria in self.settings["observations"]["filter"]:
+            selected_obs = ObservationTable()
+
+            selection["type"] = criteria["filter_type"]
+            if "inverted" in criteria and criteria["inverted"]:
+                selection["inverted"] = True
+            if selection["type"] == "sky_circle":
+                selection["frame"] = criteria["frame"]
+                selection["lon"] = Angle(criteria["lon"])
+                selection["lat"] = Angle(criteria["lat"])
+                selection["radius"] = Angle(criteria["radius"])
+                selection["border"] = Angle(criteria["border"])
+            if selection["type"] == "par_box":
+                selection["variable"] = criteria["variable"]
+                selection["value_range"] = criteria["value_range"]
+            if selection["type"] == "angle_box":
+                selection["type"] = "par_box"
+                selection["variable"] = criteria["variable"]
+                selection["value_range"] = Angle(criteria["value_range"])
+
+            if selection["type"] == "sky_circle" or selection["type"].endswith("_box"):
+                selected_obs = self.datastore.obs_table.select_observations(selection)
+            if selection["type"] == "par_value":
+                mask = (
+                    self.datastore.obs_table[criteria["variable"]]
+                    == criteria["par_value"]
+                )
+                selected_obs = self.datastore.obs_table[mask]
+            if selection["type"] == "ids":
+                obs_list = self.datastore.get_observations(criteria["obs_ids"])
+                selected_obs["OBS_ID"] = [obs.obs_id for obs in obs_list.list]
+            if selection["type"] == "all":
+                obs_list = self.datastore.get_observations()
+                selected_obs["OBS_ID"] = [obs.obs_id for obs in obs_list.list]
+            if len(selected_obs):
+                if "exclude" in criteria and criteria["exclude"]:
+                    ids.difference_update(selected_obs["OBS_ID"].tolist())
+                else:
+                    ids.update(selected_obs["OBS_ID"].tolist())
+        if len(ids):
+            self.observations = self.datastore.get_observations(ids)
+        else:
+            self.observations = Observations()
+
+    def reduce_data(self):
+        """Produce reduced data sets."""
+        self.config.validate()
```

Comment:
Remove this `self.config.validate` call in `reduce_data`?
Generic config validation should happen in Analysis init.
If specific validation is needed concerning config and data reduction (probably not?), there should be an extra method for that.

Reply:
I think we need a validation of some kind before most/all of the actions of the high-level interface (i.e. get_observations, reduce_data, optimize, etc.) In principle, the user may add/modif config settings in between these actions, so a validation of the settings is desirable. For the moment I use the generic one because until now I manage to put all the conditions in the schema, but if we consider that more complex specific validation is needed I agree this should be in an extra method.

Reply:
Well, we do have a *__validate_reduction_settings* helper function now. I needed that.